src/HotChocolate/Core/src/Execution/Extensions/ExecutionSchemaExtensions.cs

```diff
@@ -17,7 +17,24 @@ public static class ExecutionSchemaExtensions
             throw new ArgumentNullException(nameof(schema));
         }
 
-        return new ServiceCollection()
+        return schema.MakeExecutable(new ServiceCollection());
+    }
+
+    public static IRequestExecutor MakeExecutable(
+        this ISchema schema,
+        IServiceCollection services)
+    {
+        if (schema is null)
+        {
+            throw new ArgumentNullException(nameof(schema));
+        }
+
+        if (services is null)
+        {
+            throw new ArgumentNullException(nameof(services));
+        }
+
+        return services
```

Comment:
Yes, now I know I can do this in the test itself. But I got inspired by some other unit tests and then I spent almost 2 hours debugging my code and trying to find out, why the DI "loses" my service registration along the way. It turns out `.MakeExecutable()` somehow ignores call to `ISchema.AddServices()`. This was my original code.

```
// arrange
var services = new ServiceCollection();
services.AddSingleton<IBar10Resolver, Bar10Resolver>();

ISchema schema = SchemaBuilder.New()
    .AddServices(services) // this line is pretty much ignored, once .MakeExecutable() is called
    .AddGlobalObjectIdentification()
    .AddQueryType<Foo10>()
    .Create();

IRequestExecutor executor = schema.MakeExecutable();
```
I thought this change would make the situation a bit better, but it would be nice to maybe look at how we can preserve setup done by the `.AddServices()` itself.

Reply:
Reverted the changes.